src/S.bs.js

```diff
@@ -126,30 +126,30 @@ function toReason(nestedLevelOpt, error) {
     case /* UnexpectedValue */2 :
         break;
     case /* TupleSize */3 :
-        return "Expected Tuple with " + reason.expected.toString() + " items, received " + reason.received.toString();
+        return "Expected Tuple with " + reason.expected.toString() + " items, received " + reason.received.toString() + "";
     case /* ExcessField */4 :
         return "Encountered disallowed excess key \"" + reason._0 + "\" on an object. Use Deprecated to ignore a specific field, or S.Record.strip to ignore excess keys completely";
     case /* InvalidUnion */5 :
-        var lineBreak = "\n" + " ".repeat((nestedLevel << 1));
+        var lineBreak = "\n" + " ".repeat((nestedLevel << 1)) + "";
         var array = reason._0.map(function (error) {
               var reason = toReason(nestedLevel + 1, error);
               var nonEmptyPath = error.path;
               var $$location = nonEmptyPath.length !== 0 ? "Failed at " + formatPath(nonEmptyPath) + ". " : "";
-              return "- " + $$location + reason;
+              return "- " + $$location + "" + reason + "";
             });
         var reasons = Array.from(new Set(array));
-        return "Invalid union with following errors" + lineBreak + reasons.join(lineBreak);
+        return "Invalid union with following errors" + lineBreak + "" + reasons.join(lineBreak) + "";
     
   }
-  return "Expected " + reason.expected + ", received " + reason.received;
+  return "Expected " + reason.expected + ", received " + reason.received + "";
 }
 
 function toString(error) {
   var match = error.operation;
   var operation = match ? "parsing" : "serializing";
   var reason = toReason(undefined, error);
   var pathText = formatPath(error.path);
-  return "[ReScript Struct]" + " Failed " + operation + " at " + pathText + ". Reason: " + reason;
+  return "[ReScript Struct] Failed " + operation + " at " + pathText + ". Reason: " + reason + "";
```

Comment:
Here it joined to string values on compile-time, that's super cool!
(`[ReScript Struct]` and ` Failed `)